app/routes/press-releases/press-release/published.js

```diff
@@ -0,0 +1,8 @@
+import Route from '@ember/routing/route';
+
+export default class PressReleasesPressReleasePublishedRoute extends Route {
+  setupController(controller, model) {
+    const pressRelease = model.pressRelease;
+    super.setupController(controller, pressRelease);
```

Comment:
Het is niet de bedoeling in Ember om het model van een route te gaan aanpassen in de `setupController()` method. Het model dat je wil, moet gereturned worden in de `model`-hook van deze route.

Je kan de `model`-hook van de parent route best nog aanpassen zodanig dat `press-release.press-release` het ember-data record van het persbericht teruggeeft en de logica van de snapshot enkel uitgevoerd wordt in `press-release.press-release` op basis van het model van de parent-route. In deze route kan je dan gewoon hetzelfde model als de parent-route teruggeven. De logica van de snapshot is hier immers niet nodig.